code_examples/sdk_examples/src/workshop/attester/generateKeypairs.ts

```diff
@@ -1,42 +1,33 @@
 import * as Kilt from '@kiltprotocol/sdk-js'
-import {
-  blake2AsU8a,
-  keyExtractPath,
-  keyFromPath,
-  mnemonicGenerate,
-  mnemonicToMiniSecret,
-  sr25519PairFromSeed
-} from '@polkadot/util-crypto'
-import { generateAccount } from './generateAccount'
 
-// Because there is no first-class support for this class of keys,
-// we need to use a workaround to generate a key we can use for encryption/decryption.
-function generateKeyAgreement(mnemonic: string) {
-  const secretKeyPair = sr25519PairFromSeed(mnemonicToMiniSecret(mnemonic))
-  const { path } = keyExtractPath('//did//keyAgreement//0')
-  const { secretKey } = keyFromPath(secretKeyPair, path, 'sr25519')
-  return Kilt.Utils.Crypto.makeEncryptionKeypairFromSeed(blake2AsU8a(secretKey))
-}
+import { mnemonicGenerate } from '@polkadot/util-crypto'
 
-export function generateKeypairs(mnemonic = mnemonicGenerate()) {
-  const { account } = generateAccount(mnemonic)
+const signingKeyPairType = 'sr25519'
 
-  const authentication = {
-    ...account.derive('//did//0'),
-    type: 'sr25519'
-  } as Kilt.KiltKeyringPair
+export function generateKeypairs(mnemonic = mnemonicGenerate()): {
+  authentication: Kilt.KiltKeyringPair
+  keyAgreement: Kilt.KiltEncryptionKeypair
+  assertionMethod: Kilt.KiltKeyringPair
+  capabilityDelegation: Kilt.KiltKeyringPair
+} {
```

Comment:
Maybe we can do something like that? I think a function call inside the method parameters makes it harder to read.

```suggestion
export function generateKeypairs(maybeMnemonic: string | undefined): {
  authentication: Kilt.KiltKeyringPair
  keyAgreement: Kilt.KiltEncryptionKeypair
  assertionMethod: Kilt.KiltKeyringPair
  capabilityDelegation: Kilt.KiltKeyringPair
} {
  const mnemonic = maybeMnemonic || mnemonicGenerate()
```